src/main/java/com/onelogin/saml/Utils.java

```diff
@@ -146,52 +163,207 @@ public static Document loadXML(String xml) throws Exception {
+					Boolean.FALSE);
+		} catch (Throwable t) {  /* OK.  Not all parsers will support this attribute */}
+		try {
+			factory.setAttribute("http://apache.org/xml/features/disallow-doctype-decl",
+					Boolean.TRUE);
+		} catch (Throwable t) {  /* OK.  Not all parsers will support this attribute */}
+		try {
+			factory.setAttribute("http://javax.xml.XMLConstants/feature/secure-processing",
+					Boolean.TRUE);
+		} catch (Throwable t) {  /* OK.  Not all parsers will support this attribute */ }
+		try {
+			factory.setAttribute("http://apache.org/xml/features/nonvalidating/load-external-dtd",
+					Boolean.FALSE);
+		} catch (Throwable t) {  /* OK.  Not all parsers will support this attribute */}
+		try {
+			factory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+		} catch (Throwable t) {  /* OK.  Not all parsers will support this attribute */}
+
 		DocumentBuilder builder;
-		
+
 		try {
 			builder = factory.newDocumentBuilder();
 			Document doc = builder.parse(new InputSource(new StringReader(xml)));
 			// Loop through the doc and tag every element with an ID attribute as an XML ID node.
-	 		XPath xpath = XPathFactory.newInstance().newXPath();
-	 		XPathExpression expr = xpath.compile("//*[@ID]");
-	 		NodeList nodeList = (NodeList) expr.evaluate(doc, XPathConstants.NODESET);
-	 		for (int i=0; i<nodeList.getLength() ; i++) {
-	 			Element elem = (Element) nodeList.item(i);
-	 			Attr attr = (Attr) elem.getAttributes().getNamedItem("ID");
-	 			elem.setIdAttributeNode(attr, true);
-	 		}
+			XPath xpath = XPathFactory.newInstance().newXPath();
+			XPathExpression expr = xpath.compile("//*[@ID]");
+			NodeList nodeList = (NodeList) expr.evaluate(doc, XPathConstants.NODESET);
+			for (int i=0; i<nodeList.getLength() ; i++) {
+				Element elem = (Element) nodeList.item(i);
+				Attr attr = (Attr) elem.getAttributes().getNamedItem("ID");
+				elem.setIdAttributeNode(attr, true);
+			}
 			return doc;
 		} catch (Exception e) {
-			e.printStackTrace();
+			log.error("Error executing loadXML: " + e.getMessage(), e);
+		}
+		return null;
+	}
+
+	/**
+	 * This function attempts to validate an XML against the specified schema.
+	 * 
+	 * @param xml The XML document which should be validated
+	 * @param schema The schema filename which should be used
+	 * @throws Exception
+	 */
+	public static Document validateXML(Document xml, String schemaName)
+			throws Exception {
+		return validateXML(getStringFromDocument(xml), schemaName);
+	}
+
+	/**
+	 * This function attempts to validate an XML string against the specified
+	 * schema. It will parse the string into a DOM document and validate this
+	 * document against the schema.
+	 * 
+	 * @param xml The XML document which should be validated
+	 * @param schema The schema filename which should be used
+	 * @throws Exception
+	 */
+	public static Document validateXML(String xmlString, String schemaName, Boolean...debugMode)
```

Comment:
mean that are optional like in php debugMode = null, for now we are not supporting this parameter but the name method is ready to support it.